src/hotspot/share/prims/whitebox.cpp

```diff
@@ -2159,8 +2159,7 @@ WB_ENTRY(jboolean, WB_IsJVMCISupportedByGC(JNIEnv* env))
 WB_END
 
 WB_ENTRY(jboolean, WB_CanWriteJavaHeapArchive(JNIEnv* env))
-  return HeapShared::can_write()
-      && ArchiveHeapLoader::can_use(); // work-around JDK-8341371
+  return HeapShared::can_write() && ArchiveHeapLoader::can_use();
```

Comment:
Do we still need the `&& ArchiveHeapLoader::can_use()` part?

Reply:
I left it in since it may be a valuable check in case there are further changes or additions to the GCs.

Reply:
But the code does not do what its name suggests. If this is not required now, I think we should remove the can_load part so the code is not confusing.